server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -83,15 +83,20 @@ public IngestDocument(IngestDocument other) {
     }
 
     /**
-     * Constructor needed for testing that allows to create a new {@link IngestDocument} given the provided elasticsearch metadata,
-     * source and ingest metadata. This is needed because the ingest metadata will be initialized with the current timestamp at
-     * init time, which makes equality comparisons impossible in tests.
+     * Constructor to create an IngestDocument from its constituent maps
      */
-    public IngestDocument(Map<String, Object> sourceAndMetadata, Map<String, Object> ingestMetadata) {
+    IngestDocument(Map<String, Object> sourceAndMetadata, Map<String, Object> ingestMetadata) {
         this.sourceAndMetadata = sourceAndMetadata;
         this.ingestMetadata = ingestMetadata;
     }
 
+    /**
+     * Build an IngestDocument from values read via deserialization
+     */
+    public static IngestDocument fromWire(Map<String, Object> sourceAndMetadata, Map<String, Object> ingestMetadata) {
```

Comment:
I think the more typical naming would be `of`, so `IngestDocument.of(...)`. I would at least rename this method, and probably the test methods as well, eg `TestIngestDocument.ofSourceAndIngest`.